hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/BaseFileOnlyRelation.scala

```diff
@@ -52,6 +53,14 @@ class BaseFileOnlyRelation(sqlContext: SQLContext,
 
   override type FileSplit = HoodieBaseFileSplit
 
+  override lazy val mandatoryColumns: Seq[String] = {
+    if (isMetadataTable(metaClient)) {
```

Comment:
lets remove this special casing for metadata table? It's dealing with an abstraction few layers deeper than here.

Reply:
Good call, this actually not required here